gxa/src/test/java/uk/ac/ebi/atlas/tracks/TracksControllerWIT.java

```diff
@@ -1,84 +1,113 @@
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.jdbc.datasource.init.ResourceDatabasePopulator;
 import org.springframework.test.context.ContextConfiguration;
-import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
 import org.springframework.test.context.web.WebAppConfiguration;
 import org.springframework.test.web.servlet.MockMvc;
 import org.springframework.test.web.servlet.MvcResult;
 import org.springframework.test.web.servlet.setup.MockMvcBuilders;
 import org.springframework.web.context.WebApplicationContext;
-import uk.ac.ebi.atlas.configuration.WebConfig;
+import uk.ac.ebi.atlas.configuration.TestConfig;
 
+import javax.inject.Inject;
+import javax.sql.DataSource;
 import java.text.MessageFormat;
 
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
-@RunWith(SpringJUnit4ClassRunner.class)
 @WebAppConfiguration
-@ContextConfiguration(classes = WebConfig.class)
-public class TracksControllerWIT {
-    private static final MessageFormat URL_TEMPLATE =
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = TestConfig.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class TracksControllerWIT {
+    private static final MessageFormat BASELINE_URL_TEMPLATE =
             new MessageFormat("/experiments-content/{0}/tracks/{0}.{1}.genes.expressions.bedGraph");
+    private static final MessageFormat DIFFERENTIAL_URL_TEMPLATE =
+            new MessageFormat("/experiments-content/{0}/tracks/{0}.{1}.genes.log2foldchange.bedGraph");
+
+    @Inject
+    private DataSource dataSource;
 
     @Autowired
     private WebApplicationContext wac;
 
     private MockMvc mockMvc;
 
-    @Before
-    public void setUp() {
+    @BeforeAll
+    void populateDatabaseTables() {
+        ResourceDatabasePopulator populator = new ResourceDatabasePopulator();
+        populator.addScripts(new ClassPathResource("fixtures/experiment-fixture.sql"));
+        populator.execute(dataSource);
+    }
+
+    @AfterAll
+    void cleanDatabaseTables() {
+        ResourceDatabasePopulator populator = new ResourceDatabasePopulator();
+        populator.addScripts(new ClassPathResource("fixtures/experiment-delete.sql"));
+        populator.execute(dataSource);
+    }
+
+    @BeforeEach
+    void setUp() {
         this.mockMvc = MockMvcBuilders.webAppContextSetup(this.wac).build();
     }
 
     @Test
-    public void getTrackFile() throws Exception {
-        this.mockMvc.perform(get(URL_TEMPLATE.format(new Object[] {"E-MTAB-513", "g1"})))
+    void getTrackFile() throws Exception {
+        this.mockMvc.perform(get(BASELINE_URL_TEMPLATE.format(new Object[] {"E-MTAB-5423", "g391"})))
                 .andExpect(status().isOk())
                 .andReturn();
```

Comment:
Again, probably out of scope, but I don't think these `andReturn()` statements are needed.

Reply:
I removed them, while we’re at it.